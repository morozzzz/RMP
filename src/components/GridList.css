.grid-list {
    display: flex;
    justify-content: center;
}

.grid-list-content {
    max-width: 1080px;
    width: 60%;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
}
