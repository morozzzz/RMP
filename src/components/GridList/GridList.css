.grid-list {
    display: flex;
    justify-content: center;
}

.grid-list__content {
    max-width: 1080px;
    width: 45%;
    display: grid;
    grid-template-columns: repeat(3, 33.5%);
}
