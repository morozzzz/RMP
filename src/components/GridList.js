import React from 'react';
import PosterTile from './PosterTile';
import './GridList.css';

const GridList = ({ movies = [] }) => (
    <div className="grid-list">
        <div className="grid-list-content">
            {
                movies.map(movieData => <PosterTile data={movieData} key={movieData.id} />)
            }
        </div>
    </div>
);

export default GridList;
