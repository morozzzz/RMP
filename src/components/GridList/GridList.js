import React from 'react';
import PosterTile from '../PosterTile/PosterTile';
import './GridList.css';

const GridList = ({ movies = [], onItemClick }) => (
    <div className="grid-list">
        <div className="grid-list__content">
            {
                movies.map(movieData => (
                    <PosterTile data={movieData} key={movieData.id} onTileClick={onItemClick} />
                ))
            }
        </div>
    </div>
);

export default GridList;
