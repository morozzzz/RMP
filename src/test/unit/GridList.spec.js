import React from 'react';
import { shallow } from 'enzyme';
import GridList from '../../components/GridList/GridList';

jest.mock('../../components/PosterTile/PosterTile', () => () => <div id="PosterTile" />);

const moviesMock = [
    {
        id: 333339,
        title: 'Ready Player One',
        tagline: 'A better reality awaits.',
    },
    {
        id: 338970,
        title: 'Tomb Raider',
        tagline: 'Her legend begins',
    },
];


describe.skip('GridList', () => {
    it('should be rendered correctly', () => {
        const component = shallow(<GridList data={moviesMock} />);

        expect(component).toMatchSnapshot();
    });
});
