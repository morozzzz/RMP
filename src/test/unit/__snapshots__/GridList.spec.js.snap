// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridList should be rendered correctly 1`] = `
<div
  className="grid-list"
>
  <div
    className="grid-list__content"
  >
    <Component
      data={
        Object {
          "id": 333339,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
        }
      }
      key="333339"
    />
    <Component
      data={
        Object {
          "id": 338970,
          "tagline": "Her legend begins",
          "title": "Tomb Raider",
        }
      }
      key="338970"
    />
  </div>
</div>
`;
